src/core/store/external/getters.js

```diff
@@ -16,7 +20,110 @@ const balanceFiatValue = function (state, getters, rootState, rootGetters) {
   return new BigNumber(balanceInETH).times(getters.fiatValue);
 };
 
+const totalTokenFiatValue = function (state, getters, rootState, rootGetters) {
+  const tokenList = rootGetters['wallet/tokensList'];
+  const rate = state.currencyRate.data?.exchange_rate || 1;
+  if (!tokenList.length) return new BigNumber(0);
+  const totalValue = tokenList.reduce((total, currentVal) => {
+    const balance =
+      currentVal.usdBalance !== null &&
+      (currentVal.price_change_percentage_24h !== null ||
+        currentVal.market_cap !== 0)
+        ? currentVal.usdBalance
+        : 0;
+    return new BigNumber(total).plus(balance);
+  }, 0);
+  return totalValue.times(rate);
+};
+
+/**
+ * Get main currency market info
+ */
+const networkTokenUSDMarket = function (
+  state,
+  getters,
+  rootState,
+  rootGetters
+) {
+  const cgid = rootGetters['global/network'].type.coingeckoID;
+  if (cgid) {
+    const token = state.coinGeckoTokens.get(cgid);
+    if (token)
+      return {
+        value: token.current_price,
+        symbol: '$',
+        name: 'USD',
+        price_change_percentage_24h: token.price_change_percentage_24h
+      };
+  }
+  return {
+    value: 0,
+    symbol: '$',
+    name: 'USD',
+    price_change_percentage_24h: 0
+  };
+};
+
+/**
+ * Get Token info including market data if exists
+ */
+const contractToToken =
+  (state, getters, rootState, rootGetters) => contractAdress => {
+    contractAdress = contractAdress.toLowerCase();
```

Comment:
typo i think ..is it suppose to be address?

Reply:
yea so basically you can pass the `contractAddress` of the token, and itll return the complete token object including market data if that contractAddress is associated with a token in the network tokens list

Reply:
oh oh i meant cause it says contractAdress hahah not a big deal